csp/tests/test_middleware.py

```diff
@@ -82,3 +82,20 @@ def test_debug_exempt():
     response = HttpResponseServerError()
     mw.process_response(request, response)
     assert HEADER not in response
+
+
+def test_nonce_created_when_accessed():
+    request = rf.get('/')
+    mw.process_request(request)
+    nonce = request.csp_nonce()
+    response = HttpResponse()
+    mw.process_response(request, response)
+    assert nonce in response[HEADER]
+
```

Comment:
A side consideration here is what needs to be considered around caching since a different nonce value needs to be sent with every response.